assets/src/edit-story/components/canvas/displayElement.js

```diff
@@ -51,13 +52,38 @@ const BackgroundOverlay = styled.div`
   top: 0;
   left: 0;
 `;
+
 const ReplacementContainer = styled.div`
   transition: opacity 0.25s cubic-bezier(0, 0, 0.54, 1);
   pointer-events: none;
   opacity: ${({ hasReplacement }) => (hasReplacement ? 1 : 0)};
 `;
 
-function DisplayElement({ element, previewMode, page }) {
+function AnimationWrapper({ children, id, animationMode }) {
+  switch (animationMode) {
+    case 'WAAPI':
+      return (
+        <StoryAnimation.WAAPIWrapper target={id}>
+          {children}
+        </StoryAnimation.WAAPIWrapper>
+      );
+    case 'AMP':
+    default:
+      return children;
+  }
+}
```

Comment:
This isn't obvious _but_ the code to render the in-dashboard story preview and the code to generate an `amp-story` preview are totally separate.  The keyword this repo uses to divide them is `display` and `output`.  If you take a look at the import for `WithMask`, you'll notice it's coming from `/mask/display`, the equivalent `WithMask` used in an `amp-story` is pulled from `/mask/output`.  I believe every (or basically every) element rendered in a story (images, text, shapes, etc...) follow this pattern.  The `display` version creates regular html to view in the app and the `output` version creates `amp` specific tags to render the equivalent.

All this to say that `displayElement` will only be rendered in previews in the Dashboard or Editor and _not_ in an `amp-story`, so there is no need to support the `AMP` option (in fact, the `AMP` animations we're using will only run in an `amp-story` so they wouldn't work even if we tried to tag them in).  

So the supported animation modes for this wrapper should be `none` or `WAAPI`, and since it's only two (and I don't believe that'll ever change), I wonder if it should be changed to a boolean of `enableAnimations` (or something like that) and by default it's false so nothing changes, but for the dashboard previews we set it to true?

Reply:
makes sense. yea totally agree knowing that's the case